src/main/java/fi/aalto/cs/apluscourses/intellij/actions/OpenItemAction.java

```diff
@@ -58,14 +61,24 @@ public void actionPerformed(@NotNull AnActionEvent e) {
       return;
     }
 
-    SubmissionResultViewModel submission =
-        (SubmissionResultViewModel) exercisesTree.findSelected().getLevel(3);
-    if (submission == null) {
+    SelectableNodeViewModel<?> nodeViewModel = exercisesTree.getSelectedItem();
+    if (nodeViewModel == null) {
+      return;
+    }
+
+    String url;
+    if (nodeViewModel instanceof SubmissionResultViewModel) {
+      url = ((SubmissionResultViewModel) nodeViewModel).getModel().getUrl();
```

Comment:
I think we should change the name of `getUrl` in `SubmissionResult` to `getHtmlUrl`. Not only is it consistent with the other two, but it's also more correct, since it is in fact a URL that is intended to be opened in a browser and not an API end point.

After that we have three different classes with the method `getHtmlUrl`, which raises the question if they should all implement an interface with a method `getHtmlUrl`.